fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java

```diff
@@ -1194,7 +1194,7 @@ public PlanFragment visitPhysicalSchemaScan(OptExpression optExpression, ExecPla
             }
 
             // set a per node log scan limit to prevent BE/CN OOM
-            if (scanNode.getLimit() > 0) {
+            if (scanNode.getLimit() > 0 && predicates.isEmpty()) {
```

Comment:
any context for this change?

Reply:
If the query has predicates, should not push down limit to scannode

Reply:
The logic here is oversimplified, but it is practical. There are some conditions that the limit should be push down, i.e. 
```
select log from be_logs where log rlike 'xxx'
```
If all predicate is also pushed down, we can push down limit, but the logic to check all predicates are pushed down is complex, so it's not implemented.